tests/integration/test_kubernetes_master_integration.py

```diff
@@ -21,15 +22,51 @@ def _check_status_messages(ops_test):
             assert unit.workload_status_message == message
 
 
+async def _retrieve_url(url, target_file):
+    chunk_size = 16000
+    async with aiohttp.ClientSession() as session:
+        async with session.get(url) as resp:
+            with target_file.open("wb") as fd:
+                async for chunk in resp.content.iter_chunked(chunk_size):
+                    fd.write(chunk)
+
+
 @pytest.mark.abort_on_fail
 async def test_build_and_deploy(ops_test):
+    log.info("Retrieve CNI...")
+    cni_amd64 = ops_test.tmp_path / "cni-amd64.tgz"
+    await _retrieve_url(CNI_AMD64_URL, cni_amd64)
```

Comment:
Long-term, I would really like to see the integration tests build the CNI resources rather than pulling a pre-built one.

Reply:
totally agree, i could make that change pretty quickly